mypy 1.13.0 fails in `branch-3.7`

Fails with:
```py
mypy 1.13.0 (compiled: yes)
src/bokeh/palettes.py: note: In function "interp_palette":
src/bokeh/palettes.py:1676:23: error: Argument 1 to "RGB" has incompatible type
"*tuple[unsignedinteger[_8Bit], unsignedinteger[_8Bit], unsignedinteger[_8Bit], floating[Any]]";
expected "int"  [arg-type]
        return tuple(RGB(*args).to_hex() for args in zip(r, g, b, a))
                          ^~~~
src/bokeh/palettes.py:1676:23: error: Argument 1 to "RGB" has incompatible type
"*tuple[unsignedinteger[_8Bit], unsignedinteger[_8Bit], unsignedinteger[_8Bit], floating[Any]]";
expected "float"  [arg-type]
        return tuple(RGB(*args).to_hex() for args in zip(r, g, b, a))
                          ^~~~
Found 2 errors in 1 file (checked 656 source files)
```

--- a/src/bokeh/colors/color.py
+++ b/src/bokeh/colors/color.py
@@ -25,12 +25,13 @@ import colorsys
 from abc import ABCMeta, abstractmethod
 from math import sqrt
 from re import match
-from typing import TYPE_CHECKING, TypeAlias
+from typing import TYPE_CHECKING, TypeAlias, cast
 
 # Bokeh imports
 from ..core.serialization import AnyRep, Serializable, Serializer
 
 if TYPE_CHECKING:
+    import numpy as np
     from typing_extensions import Self
 
 #-----------------------------------------------------------------------------
@@ -207,7 +208,12 @@ class RGB(Color):
 
     '''
 
-    def __init__(self, r: int, g: int, b: int, a: float = 1.0) -> None:
+    r: int
+    g: int
+    b: int
+    a: float
+
+    def __init__(self, r: int | np.integer, g: int | np.integer, b: int | np.integer, a: float | np.floating = 1.0) -> None:
         '''
 
         Args:
@@ -224,10 +230,10 @@ class RGB(Color):
                 An alpha value for this color in [0, 1] (default: 1.0)
 
         '''
-        self.r = r
-        self.g = g
-        self.b = b
-        self.a = a
+        self.r = cast(int, r)
+        self.g = cast(int, g)
+        self.b = cast(int, b)
+        self.a = cast(float, a)
 
     def copy(self) -> RGB:
         ''' Return a copy of this color value.

--- a/src/bokeh/palettes.py
+++ b/src/bokeh/palettes.py
@@ -1602,7 +1602,7 @@ def varying_alpha_palette(color: str, n: int | None = None, start_alpha: int = 0
         seq[str] : a sequence of hex RGBA color strings
 
     Raises:
-        ValueError if ``color`` is not recognisable as a string name or hex
+        ValueError if ``color`` is not recognizable as a string name or hex
             RGB(A) string, or if ``start_alpha`` or ``end_alpha`` are outside
             the range 0 to 255 inclusive.
 

